### PR TITLE
[DS-4136] Master port: Improve OAI import performance for a large install

### PR DESCRIPTION
**Pull request by kshepherd**
Master port of 6.x PR #2320 which is already merged (as is the 5.x port)
JIRA: https://jira.duraspace.org/browse/DS-4136

### CODE REVIEW

**Review by terrywbrady — APPROVED** [review]

File: dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java [hunk 1]
```diff
@@ -54,15 +54,15 @@
 import org.dspace.content.MetadataValue;
 import org.dspace.content.factory.ContentServiceFactory;
 import org.dspace.content.service.ItemService;
-import org.dspace.core.ConfigurationManager;
 import org.dspace.core.Constants;
 import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService; //added
```
Comment (tdonohue) [inline dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java]: Looks like an unnecessary comment here.

**Review by tdonohue — APPROVED** [review]
👍 Looks good to me, it's almost an exact copy of #2320, except for an unnecessary trailing comment (see inline note below).  Nonetheless, the trailing comment isn't harmful, so +1 from me.

File: dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java [hunk 2]
```diff
@@ -300,12 +304,20 @@ private int index(Iterator<Item> iterator)
                     log.error(ex.getMessage(), ex);
                 }
                 i++;
-                if (i % 100 == 0) {
+                if (i % 1000 == 0) {
```
Comment (mwoodiupui) [inline dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java]: Note that we get the message twice when oai.import.batch.size is defaulted.  Cosmetic.  But do we really need this?

**Review by mwoodiupui — APPROVED** [review]
+1 by inspection.  Looks well, aside from a quibble or two.

File: dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java (hunk 2 above)
Comment (kshepherd) [reply dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java]: yes good point - i was going to leave this as is to be a straight 6.x port, but might as well tidy up here then do another small change to 6.x to match it
so now, that test is `if(i % 1000 == 0 && batchSize != 1000)`

File: dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java (hunk 1 above)
Comment (kshepherd) [reply dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java]: i've removed this comment now, thanks

File: dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java (hunk 2 above)
Comment (mwoodiupui) [reply dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java]: Sounds well, thanks!

**Pull request merged**